### PR TITLE
EZP-24061: As an editor, I want to be able to select several contents in the universal discovery widget

### PR DESCRIPTION
**Pull request by dpobel**
JIRA: https://jira.ez.no/browse/EZP-24061

# Description

The patch changes the universal discovery widget to allow a multiple selection making it suitable to fill a relation list field for instance.
So now, when the universal discovery widget is configured to allow several contents to be selected, the selected view displays a *Choose this content* button and when using it the content is put in the confirm list. Of course, after the validation of the selection, it's possible to use the selected content list in any way (in the screenast, I only display the name of the selected contents).

Screencast: http://youtu.be/Zc0X0QVqV4Q

## Tasks

* [x] Replace the `selectionMode` attribute by a boolean flag
* [x] Actual support for a multiple content selection
* [x] Display a part of the selection in the bottom left corner
* [x] Do not allow to select the same content twice
* [x] Animation on content selection
* [x] Disable the choose button if the content is already selected

# Tests

manual + unit tests

### CODE REVIEW

**Comment by dpobel** [comment]
ping @StephaneDiot @yannickroger

**Comment by andrerom** [comment]
Nice :)
+1

File: Resources/public/js/views/ez-universaldiscoveryview.js [hunk 1]
```diff
@@ -373,10 +507,10 @@ YUI.add('ez-universaldiscoveryview', function (Y) {
 
             /**
              * The current selection of the discovery. This selection is
-             * providded to the contentDiscovered event handler in the event
-             * facade. Depending on the selectionMode and on the user action,
-             * the selection is either null or an object (selectionMode to
-             * 'single') or an array (selectionMode to 'multiple').
+             * provided to the contentDiscovered event handler in the event
+             * facade. Depending on the `multiple` flag and on the user action,
+             * the selection is either null or an object (`multiple` to
```
Comment (bdunogier) [inline Resources/public/js/views/ez-universaldiscoveryview.js]: "multiple *set* to false"

File: Resources/public/js/views/universaldiscovery/ez-universaldiscoveryconfirmedlistview.js [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+         */
+        _hasConfirmedList: function () {
+            return !!this.get('confirmedList');
+        },
+    }, {
+        ATTRS: {
+            /**
+             * The current confirmed list. It's an array containing one or
+             * several content structure (ie an object with a content, a
+             * location and a content type models or null.
```
Comment (bdunogier) [inline Resources/public/js/views/universaldiscovery/ez-universaldiscoveryconfirmedlistview.js]: Missing closing parenthesis.

File: Resources/public/js/views/universaldiscovery/ez-universaldiscoveryselectedview.js [hunk 3]
```diff
@@ -67,7 +134,30 @@ YUI.add('ez-universaldiscoveryselectedview', function (Y) {
              */
             contentStruct: {
                 value: null,
-            }
+            },
+
+            /**
+             * Flag indicating whether a confirm button is needed or not.
```
Comment (bdunogier) [inline Resources/public/js/views/universaldiscovery/ez-universaldiscoveryselectedview.js]: Shouldn't it be named "needsConfirmButton", or "requiresConfirmButton" then ?

**Comment by bdunogier** [comment]
+1, with remarks (mostly suggestions/questions).

**Comment by StephaneDiot** [comment]
I'm late, but +1 FWIW.

**Comment by bdunogier** [comment]
Nice !

**Pull request merged**